adodb.inc.php

```diff
@@ -5444,6 +5468,28 @@ function ADONewConnection($db='') {
 			define('ADODB_ASSOC_CASE', ADODB_ASSOC_CASE_NATIVE);
 		}
 
+		/*
+		* If we have a pdo and drivern in format \\, we
+		* need to remove the driver portion of the file
+		*/
+		$pdoExtension = '';
+		$pdoSplit 	  = explode('\\',$db);
```

Comment:
I think that as it stands, provision of the DSN on the newADOcommand line is no longer supported. Not to say that the processing of the parameter should be super robust.  I did note it in the Quick upgrade notes.

Reply:
> provision of the DSN on the newADOcommand line is no longer supported 
> I did note it in the Quick upgrade notes.

I missed that bit.
Is there a particular reason for dropping this functionality ? It could be fairly easily maintained by adapting the PDO-specific logic to use the $pdoExtension variable:
```diff
diff --git a/adodb.inc.php b/adodb.inc.php
index c5e30f09..6ff88c0b 100644
--- a/adodb.inc.php
+++ b/adodb.inc.php
@@ -5547,17 +5547,11 @@ class ADORecordSet implements IteratorAggregate {
                $dsna['host'] = '';
            }
 
-           if (strncmp($origdsn,'pdo',3) == 0) {
-               $sch = explode('_',$dsna['scheme']);
-               if (sizeof($sch)>1) {
-                   $dsna['host'] = isset($dsna['host']) ? rawurldecode($dsna['host']) : '';
-                   if ($sch[1] == 'sqlite') {
-                       $dsna['host'] = rawurlencode($sch[1].':'.rawurldecode($dsna['host']));
-                   } else {
-                       $dsna['host'] = rawurlencode($sch[1].':host='.rawurldecode($dsna['host']));
-                   }
-                   $dsna['scheme'] = 'pdo';
-               }
+           if ($pdoExtension) {
+               $host = $pdoExtension
+                   . ($pdoExtension == 'sqlite' ? ':' : ':host=')
+                   . rawurldecode($dsna['host'] ?? '');
+               $dsna['host'] = rawurlencode($host);
            }
 
            $db = @$dsna['scheme'];
```

Reply:
Two things on future issues, 

1. The ''_'' character can't be used if using off-the-shelf PSR style autoloaders, we'll probably need a project in the future to remove them from classnames. Might as well start somewhere.
2.   To be honest, I don't particularly like fact that you can assign a DSN through here (and not any other type of parameter). It feels very unstructured, and I'd certainly not do it if I did the original design.